lib/trino-plugin-toolkit/src/main/java/io/trino/plugin/base/util/ExceptionUtil.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.base.util;
+
+import io.trino.spi.ErrorCodeSupplier;
+import io.trino.spi.TrinoException;
+
+import java.util.Optional;
+
+import static com.google.common.base.Throwables.getCausalChain;
+
+public final class ExceptionUtil
```

Comment:
https://github.com/trinodb/trino/pull/21210 is going to introduce `Exceptions` class. Let's use the same name.